src/v/ssx/thread_worker.h

```diff
@@ -64,9 +64,19 @@ class worker_task final : public task_base {
     }
 
     void process(ss::alien::instance& alien, ss::shard_id shard) final {
-        auto f = futurator::invoke(_func);
-        ss::alien::run_on(alien, shard, [this, f{std::move(f)}]() mutable {
-            f.forward_to(std::move(_promise));
+        try {
+            set_value(alien, shard, _func());
+        } catch (...) {
+            set_exception(alien, shard, std::current_exception());
+        };
+    }
+
+    void set_value(
+      ss::alien::instance& alien,
+      ss::shard_id shard,
+      typename futurator::value_type v) {
```

Comment:
nit: taking `futurator::value_type` looks ok, but it seems noisy since we are just passing `result_t<func>`, but the value_type is something like `using value_type = internal::future_stored_type_t<T SEASTAR_ELLIPSIS>;` whatever that is.

Reply:
Used `std::invoke_result_t` instead.